python/tvm/relay/to_onnx.py

```diff
@@ -0,0 +1,422 @@
+        topologically sorted Relay Node entry list
+    """
+
+    def __init__(self, name, node_list, params):
+        self._name = {}
+        self._mc = ModelContainer(name)
+        self._node_list = node_list
+        self._params = params
+
+    def convert_to_onnx(self):
+        """ Loop through topologically sorted list of Relay nodes and generate a ONNX model"""
+        for idx, node_entry in enumerate(self._node_list):
+            out_idx = idx
+            node = node_entry['node']
+            if isinstance(node, Call):
+                self._add_node(node_entry, idx)
+            elif isinstance(node, Var):
+                self._add_input(node_entry, idx)
+            elif isinstance(node, Constant):
+                self._add_constant_input(node_entry, idx)
+            elif isinstance(node, (TupleGetItem, Tuple)):
+                out_idx = idx - 1  # Need to work on this. No equivalent ONNX operator found yet
+            else:
+                raise NotImplementedError("Relay Node of type {0} is not "
+                                          "implemented yet".format(type(node)))
+
+            if idx == len(self._node_list) - 1:
+                self._add_output(self._node_list[out_idx], out_idx)
+
+        polished_model = self._mc.make_model()
+        #polished_model = onnx.utils.polish_model(model)
+        return polished_model
+
+    def _tuple_to_name(self, input):
+        """convert tuple of node indexes to string"""
+        return 'node_{0}'.format(input[0])
+
+    def _add_node(self, node_entry, idx):
+        """Convert Relay operator node to ONNX opeartor and add it to container nodes list"""
+        if node_entry['op'] not in relay_to_onnx_op_mapping:
+            ops.add(node_entry['op'])
+            return
+            # raise NotImplementedError("Currently the operator '{0}' is "
+            #                           "not supported.".format(node_entry['op']))
+
+        converter = relay_to_onnx_op_mapping[node_entry['op']]()
+        node_entry['output_names'] = [self._tuple_to_name([idx, 0, 0])]
+        node_entry['input_names'] = []
+        for input_idx_tuple in node_entry['inputs']:
+            if self._node_list[input_idx_tuple[0]]['name']:
+                node_entry['input_names'].append(self._node_list[input_idx_tuple[0]]['name'])
+            else:
+                node_entry['input_names'].append(self._tuple_to_name(input_idx_tuple))
+
+
+        # intermediate_tensor_name = self._tuple_to_name([idx, 0, 0])
+        # intermediate_layer_value_info = onnx.helper.ValueInfoProto()
+        # intermediate_layer_value_info.name = intermediate_tensor_name
+        # self._mc.add_outputs([intermediate_layer_value_info])
+
+        converter.convert(node_entry, self._mc, self._node_list)
+
+    def _add_params(self, node_entry, idx):
+        """Add param value to initializer and name to inputs"""
+        param_name = node_entry['name']
+        value = self._params[param_name]
```

Comment:
Let's assert the existence of `param_name` in `self._params`

Reply:
Added.